### PR TITLE
Turn on esnext

### PR DESCRIPTION
**Pull request by adrianmoisey**
Since we're doing it all over the place such as https://github.com/yola/yositecomponentsjs/pull/18#issuecomment-137802932

### CODE REVIEW

File: .jshintrc [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {
   "curly": true,
+  "esnext": true,
```
Comment (Hkly) [inline .jshintrc]: Tiny nitpick, if this is in alphabetical order, it should go after `eqeqeq`

Comment (adrianmoisey) [inline .jshintrc]: Doh! I tried for that but missed. Fixing...

**Comment by DevShep** [comment]
I don't think we want this to be part of core jsgreat. Not everything will use es6 (especially SBUI), and we'd want snitch to complain about es6 in repos that don't specifically subscribe to es6.

Think it should be `0.1.6-es6` or something of that nature.

File: .jshintrc (hunk 1 above)
Comment (Hkly) [inline .jshintrc]: Sometimes I still have to sing the alphabet to figure it out... :flushed: haha

**Comment by DevShep** [comment]
> Think it should be 0.1.6-es6 or something of that nature.

This also means we'd need an es6 specific branch.

**Comment by adrianmoisey** [comment]
> I don't think we want this to be part of core jsgreat. Not everything will use es6 (especially SBUI), and we'd want snitch to complain about es6 in repos that don't specifically subscribe to es6.

This is what I was unsure of.
lintreview currently has a default config for jshint that is uses: https://github.com/yola/lintreview/blob/master/lintrc_defaults.ini#L2

We essentially want to support two config files. So I suppose that means maintaining and installing two different jshint config files. Then all of the ES6 projects will need to have a config specified in their `.lintrc` file.

**Comment by Hkly** [comment]
Oo yea that makes sense...

**Comment by DevShep** [comment]
Superseded by https://github.com/yola/jsgreat/pull/20

**Comment by stefanor** [comment]
Close this and delete the branch, then?

**Pull request closed without merging**